.github/workflows/build-and-test-snap.yml

```diff
@@ -68,8 +70,27 @@ jobs:
           name: logs
           path: tests/*.log
 
+  publish-amd64:
+    needs: [build-amd64, test-amd64]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download locally built snap
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ env.ARTIFACT_AMD64 }}
+
+      - uses: snapcore/action-publish@v1
+        env:
+          SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
+        with:
+          snap: ${{ needs.build-amd64.outputs.snap }}
+          release: edge/ieng-1057
```

Comment:
Based on the current triggers, this is going to publish both on PRs and after merged into main. The `latest/edge` release should only happen on the main branch. Optionally, the release can happen from PRs into a store branch, e.g. `latest/edge/pr-61` for the current PR.